### PR TITLE
Batch backports to 6.0.x 

### PR DESCRIPTION
**Pull request by jlucovsky**
Continuation of #9413 

Batch backports to 6.0.x

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) tickets:
- [6045](https://redmine.openinfosecfoundation.org/issues/6045)
- [6248](https://redmine.openinfosecfoundation.org/issues/6248)
- [6213](https://redmine.openinfosecfoundation.org/issues/6213)
- [6227](https://redmine.openinfosecfoundation.org/issues/6227)

Describe changes:
-

Updates:
- Removed extra includes from suricata.c

### Provide values to any of the below to override the defaults.

To use a pull request use a branch name like `pr/N` where `N` is the
pull request number.

Alternatively, `SV_BRANCH` may also be a link to an
OISF/suricata-verify pull-request.

```
SV_REPO=
SV_BRANCH=
SU_REPO=
SU_BRANCH=
LIBHTP_REPO=
LIBHTP_BRANCH=
```

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 15847

**Review by victorjulien — APPROVED** [review]

File: src/suricata.c [hunk 1]
```diff
@@ -511,19 +516,17 @@ static void SetBpfStringFromFile(char *filename)
...
+    SCStat st;
     FILE *fp = NULL;
     size_t nm = 0;
 
-#ifdef OS_WIN32
-    if(_stat(filename, &st) != 0) {
-#else
-    if(stat(filename, &st) != 0) {
-#endif /* OS_WIN32 */
+    fp = fopen(filename, "r");
```
Comment (victorjulien) [inline src/suricata.c:523]: bad merge, as fopen is already done below. Have a fix

**Review by victorjulien — CHANGES_REQUESTED** [review]
fixes needed, doing them myself

**Comment by victorjulien** [comment]
replaced by #9457

**Pull request closed without merging**